toolcraft@0.0.12: bundled @poe-code/task-list imports unbundled @poe-code/process-runner — runCLI fails at module load

## Summary

`toolcraft@0.0.12` is unusable as published: importing anything from `toolcraft/cli` crashes at module load because a transitive `@poe-code/process-runner` is neither bundled in the tarball nor published to npm.

## Repro

```bash
mkdir t && cd t && bun init -y
bun add toolcraft@0.0.12
echo 'import { runCLI } from "toolcraft/cli"; console.log(typeof runCLI);' > x.ts
bun x.ts
```

```
error: Cannot find module '@poe-code/process-runner' from
'.../node_modules/.bun/toolcraft@0.0.12/node_modules/toolcraft/node_modules/@poe-code/task-list/dist/backends/gh-issues-client.js'
```

## Evidence

`toolcraft@0.0.12`'s `package.json` lists `@poe-code/task-list` in `bundleDependencies` (good — it's in the tarball at `package/node_modules/@poe-code/task-list/`).

But the bundled `@poe-code/task-list@0.0.1`'s own `package.json` declares:

```json
"dependencies": {
  "@poe-code/file-lock": "*",
  "@poe-code/process-runner": "*",
  "yaml": "*"
}
```

`@poe-code/process-runner` is:

- **not** in toolcraft's bundle (`ls package/node_modules/@poe-code/` → no `process-runner`)
- **not** published on npm — `curl -sI https://registry.npmjs.org/@poe-code/process-runner` returns 404

It is imported eagerly at the top of `@poe-code/task-list/dist/backends/gh-issues-client.js`:

```js
import { createHostRunner } from "@poe-code/process-runner";
```

…and that file is reachable from toolcraft's CLI entrypoint, so any `import "toolcraft/cli"` crashes.

The same `*`-version pattern means several other transitive deps (`tiny-mcp-client`, `auth-store`) only work because they happen to be bundled — `process-runner` slipped through.

## Suggested fix

Any of:

1. Add `@poe-code/process-runner` to toolcraft's `bundleDependencies` so the tarball is self-contained.
2. Publish `@poe-code/process-runner` (and the other `@poe-code/*` workspace packages) to npm so installers can resolve them normally — and prefer real semver ranges over `*`.
3. Make the `gh-issues-client` backend in `@poe-code/task-list` lazy-load `@poe-code/process-runner` so it's only required when that backend is actually used.

For context: this blocked an attempt to migrate from `poe-code/cmdkit` (dropped from `poe-code` exports at 3.0.188) to standalone `toolcraft`. We're currently pinned at `poe-code@3.0.187` because of this.

--- a/.github/workflows/release-toolcraft.yml
+++ b/.github/workflows/release-toolcraft.yml
@@ -49,6 +49,8 @@ jobs:
           cd ../toolcraft-openapi && npm pkg set "dependencies.toolcraft=^${NEXT}"
       - name: Build toolcraft packages and dependencies
         run: npx turbo build --filter=toolcraft-openapi...
+      - name: Verify standalone install resolves all bundled imports
+        run: npm run smoke:toolcraft-standalone
       - name: Publish toolcraft-schema
         working-directory: packages/toolcraft-schema
         run: npm publish --provenance --access public

--- a/packages/toolcraft/package.json
+++ b/packages/toolcraft/package.json
@@ -35,8 +35,8 @@
     "test": "cd ../.. && vitest run packages/toolcraft/src",
     "test:unit": "cd ../.. && vitest run packages/toolcraft/src",
     "lint": "cd ../.. && eslint packages/toolcraft/src --ext ts && tsc -p packages/toolcraft/tsconfig.json --noEmit",
-    "prepack": "node ../../scripts/manage-bundled-workspace-deps.mjs prepare . @poe-code/design-system @poe-code/agent-mcp-config @poe-code/agent-human-in-loop @poe-code/task-list @poe-code/file-lock @poe-code/agent-defs @poe-code/config-mutations tiny-mcp-client mcp-oauth auth-store",
-    "postpack": "node ../../scripts/manage-bundled-workspace-deps.mjs cleanup . @poe-code/design-system @poe-code/agent-mcp-config @poe-code/agent-human-in-loop @poe-code/task-list @poe-code/file-lock @poe-code/agent-defs @poe-code/config-mutations tiny-mcp-client mcp-oauth auth-store"
+    "prepack": "node ../../scripts/manage-bundled-workspace-deps.mjs prepare . @poe-code/design-system @poe-code/agent-mcp-config @poe-code/agent-human-in-loop @poe-code/task-list @poe-code/file-lock @poe-code/agent-defs @poe-code/config-mutations @poe-code/process-runner tiny-mcp-client mcp-oauth auth-store",
+    "postpack": "node ../../scripts/manage-bundled-workspace-deps.mjs cleanup . @poe-code/design-system @poe-code/agent-mcp-config @poe-code/agent-human-in-loop @poe-code/task-list @poe-code/file-lock @poe-code/agent-defs @poe-code/config-mutations @poe-code/process-runner tiny-mcp-client mcp-oauth auth-store"
   },
   "dependencies": {
     "@clack/core": "^1.0.0",
@@ -45,6 +45,10 @@
     "chalk": "^5.6.2",
     "commander": "^14.0.3",
     "console-table-printer": "^2.15.0",
+    "jose": "^6.1.2",
+    "jsonc-parser": "^3.3.1",
+    "mustache": "^4.2.0",
+    "smol-toml": "^1.3.0",
     "tiny-stdio-mcp-server": "^0.1.0",
     "yaml": "^2.8.2"
   },
@@ -67,6 +71,7 @@
     "@poe-code/file-lock",
     "@poe-code/agent-defs",
     "@poe-code/config-mutations",
+    "@poe-code/process-runner",
     "tiny-mcp-client",
     "mcp-oauth",
     "auth-store"
@@ -79,6 +84,7 @@
     "@poe-code/file-lock": "*",
     "@poe-code/agent-defs": "*",
     "@poe-code/config-mutations": "*",
+    "@poe-code/process-runner": "*",
     "tiny-mcp-client": "*",
     "mcp-oauth": "*",
     "auth-store": "*"

--- a/scripts/verify-toolcraft-standalone.mjs
+++ b/scripts/verify-toolcraft-standalone.mjs
@@ -105,7 +105,14 @@ function runConsumerSmoke(projectDir, tarballs) {
     [
       "--input-type=module",
       "--eval",
-      ['await import("toolcraft");', 'await import("toolcraft/human-in-loop");'].join("\n")
+      [
+        'await import("toolcraft");',
+        'await import("toolcraft/cli");',
+        'await import("toolcraft/mcp");',
+        'await import("toolcraft/mcp-proxy");',
+        'await import("toolcraft/sdk");',
+        'await import("toolcraft/human-in-loop");'
+      ].join("\n")
     ],
     { cwd: projectDir, stdio: "inherit" }
   );

